### PR TITLE
[PARTI-406] Skip catalog ingestion if catalog is empty

### PR DESCRIPTION
**Pull request by KagenLH**
# PR Description

This PR adds the following changes:

- When doing a catalog ingestion and all of the following fields in the payload are empty, exits early (after fetching and transformation):
  - `items`
  - `variations`
  - `groups` 
- Adds null as a possible resolved promise return type for the `ingest` method (this will likely need to be handled in repos that consume the library).

# How to test

## 🧪 What should be tested?

As a reviewer, you should test:

- Catalog ingestion exits early when the fetched payload is empty.

## 📝 How to test?

Reproduce the following steps to test:

1. Pull the branch, adjust your `dev.ts` to capture and console.log the return value of the ingestion. Try with an empty payload and with non-empty.

### CODE REVIEW

File: src/catalogIngestor/index.ts [hunk 1]
```diff
@@ -27,18 +27,22 @@ export class CatalogIngestor<TType extends CatalogIngestionType> {
...
 
     payload = await getData();
 
     logMemoryUsage(
-      `Finished fetching data. Got ${payload.data.groups.length} groups, ${payload.data.items.length} items and ${payload.data.variations.length} variations.`
+      `Finished fetching data. Got ${payload.data.groups.length} groups, ${payload.data.items.length} items and ${payload.data.variations.length} variations.`,
     );
 
+    if (this.payloadIsEmpty(payload)) {
+      return null;
```
Comment (pedro-lb) [inline src/catalogIngestor/index.ts:43]: **Issue:** I think we should always return an object. We may want to add other properties in the future and this `null` will make it harder for us to do that.

Inside the object, we can set `taskId` to be null.

What do you think?

**Review by pedro-lb — CHANGES_REQUESTED** [review]
Thanks for the work here @KagenLH!

Left a small suggestion regarding the return types, but I do think that it's a blocker. Let's fix it before we proceed!

File: src/catalogIngestor/index.ts (hunk 1 above)
Comment (KagenLH) [reply src/catalogIngestor/index.ts]: That makes good sense. Changed with https://github.com/Constructor-io/constructorio-catalog-ingestor/pull/206/commits/c85c44490b99509272427bda0c706bbfd9c03603

**Review by pedro-lb — APPROVED** [review]
LGTM! Thanks for the awesome work here @KagenLH! 🔥🔥🔥
 
I think we're ready to 🚢 it!

**Pull request merged**